torch/csrc/jit/codegen/cuda/ir_interface_nodes.h

```diff
@@ -181,25 +181,160 @@ namespace ir_utils {
 class TVDomainGuard;
 }
 
-// TensorView is our primitive Tensor Type used in code generation. It can be
-// thought of as representing physical memory, however, its dimensionality is
-// modifed as split/merge/computeAt functions are called. The history of
-// these transformations are kept and used for generating actual code referncing
-// physical memory. Generally when users are thinking of code generation in
-// reference to a Tensor, this is the class they should be interacting with.
-//
-// The reason we need both TensorView and TensorDomain is that we need to have a
-// record of both what is being computed and how it is being computed. For
-// example we may have the operation: TV3[I, J, K] = TV2[I, J, K] + TV1[I, J, K]
-// The mathematical operations here are on the tensor views TV1, TV2, and TV3.
-// This operation is a pointwise operation. To compute this pointwise operation
-// we iterate over the 3D TensorDomain [I, J, K], where K is the fastest
-// changing dimension.
-//
-// TODO: Need to work on the const model for TensorView, making all functions
-// that should be const, const. Gave this a try but expanded really quickly.
-// getComputeAtAxis not being const because it can return a TV that some expect
-// to be non-const is the biggest headache.
+/**
+ * TensorViewOptions class is intended to be used with
+ * TensorView::makeTensor(TensorviewOptions tvo). TensorViewOptions allows users
+ * to easily set the properties of the TensorView to be constructed. This
+ * includes sizes, contiguity, number of dimensions, and type.
+ *
+ */
+class TensorViewOptions {
+ public:
+  /// Set the number of dimensions of the tensor
+  TensorViewOptions nDims(int ndims) const {
+    TensorViewOptions tvo = *this;
```

Comment:
I can see this pattern has been copied from TensorOptions, but I see no reason why we can't be more concise:

```
ndims_ = ndims;
return *this;
```